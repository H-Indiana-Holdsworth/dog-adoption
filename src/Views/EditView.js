import React from 'react';
import DogForm from '../Components/DogForm';
import Header from '../Components/Header';
import { useState, useEffect } from 'react';
import { getDogsById, updateDog } from '../services/service-dogs';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';

export default function EditView() {
  const [dog, setDog] = useState({});
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getDogsById(params.id);
      setDog(resp);
      setLoading(false);
    };
    fetchData();
  }, [params.id, loading]);

  if (loading) {
    return <h3>Loading...</h3>;
  }

  const updateDogState = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await updateDog(dog);
    alert(`${dog.name} was successfully updated!`);
  };

  return (
    <div>
      <Header />
      <DogForm {...dog} updateDogState={updateDogState} handleSubmit={handleSubmit} />
    </div>
  );
}
