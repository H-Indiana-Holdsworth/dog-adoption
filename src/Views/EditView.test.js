import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import EditView from './EditView';

test('EditView should render a name', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/17/edit']}>
      <Route exact path="/dogs/:id/edit" component={EditView} />
    </MemoryRouter>
  );

  await screen.findByDisplayValue('sterinway');

  expect(container).toMatchSnapshot();
});
