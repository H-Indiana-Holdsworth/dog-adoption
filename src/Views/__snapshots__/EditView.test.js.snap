// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditView should render header 1`] = `
<div>
  <div>
    <header>
      <a
        class="home-link"
        href="/"
      >
        Home
      </a>
    </header>
    <div
      class="body"
    >
      <form>
        <label>
          Name:
          <input
            name="name"
            type="text"
            value="Lil Mark"
          />
        </label>
        <label>
          Image URL:
          <input
            name="image"
            type="text"
            value="https://placedog.net/500?id=41"
          />
        </label>
        <label>
          Bio:
          <input
            class="bio"
            name="bio"
            type="text"
            value="Hide your Llamas and their milk, Lil Mark loves that stuff"
          />
        </label>
        <label>
          Breed:
          <input
            name="breed"
            type="text"
            value="Pug"
          />
        </label>
        <label>
          Age:
          <input
            name="age"
            type="text"
            value="6"
          />
        </label>
        <button>
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
